### ISSUE TITLE
Feature Request: Proxy requests through a specified server

### ISSUE THREAD

**Issue by BenDavidAaron** [jmhale/okta-awscli#113]
**Is your feature request related to a problem? Please describe.**
My Okta org requires MFA when connecting from outside our network (eg: on an EC2 instance), making `okta-awscli` depend on answering a push notification on my phone. This works but  when CICD jobs are triggered periodically I have to answer notifications that I didn't originate (bad security practice). 

**Describe the solution you'd like**
Allow users to proxy requests to Okta for secure credentials through a specified server in their organization's network to avoid MFA requirements when possible (after an appropriate warning). This would reduce friction in automated awscli uses such as CICD systems. 

Add argument: `okta-awscli --okta-profile $profile_name --proxy-server 1.2.3.4:1234`

**Describe alternatives you've considered**
Currently I use Okta app based MFA, and it has the limitations described above. 

**Additional context**
Any suggestions are appreciated. Thanks for considering!

**Comment by arahayrabedian** [comment]
Dunno if it helps, but was skimming through the issues and saw this one.
If you play your cards right, _maybe_ the technique outlined in the links below in unison could help you?

https://github.com/boto/boto3/issues/338 and https://2.python-requests.org/en/master/user/advanced/#proxies

TL;DR - set proxies env var and see if that helps

**Comment by BenDavidAaron** [comment]
I'll have to give that a try once my org sets up an on-prem proxy. Thanks!

**Comment by jmhale** [comment]
Sorry for the late reply here. @arahayrabedian's suggestion makes sense to me. Since `okta-awscli` is just using `boto3` and `requests` under the hood, setting `HTTPS_PROXY` should be sufficient to cover everything.

You could even do it as a one-liner like `HTTPS_PROXY=https://1.2.3.4:5678; okta-awscli ...`

Going to close this for now, but if people find that the proxy envvar solution doesn't work, we can revisit building that into the project itself.
